models/iosxr/acls/iosxr_acls.yaml

```diff
@@ -0,0 +1,473 @@
+                          type: bool
+                        urg:
+                          description: Match on the URG bit
+                          type: bool
+                    igmp:
+                      description: Internet Group Management Protocol (IGMP) settings.
+                      type: dict
+                      suboptions:
+                        dvrmp:
+                          description: Match Distance Vector Multicast Routing Protocol
+                          type: bool
+                        host_query:
+                          description: Match Host Query
+                          type: bool
+                        host_report:
+                          description: Match Host Report
+                          type: bool
+                        pim:
+                          description: Match Protocol Independent Multicast
+                          type: bool
+                        trace:
+                          description: Multicast trace
+                          type: bool
+                        mtrace:
+                          description: Match mtrace
+                          type: bool
+                        mtrace_response:
+                          description: Match mtrace response
+                          type: bool                
+                dscp:
+                  description: 
+                    - Match packets with given dscp value.
+                  type: str
+                fragments:
+                  description:
+                    - Check non-intial fragments.
+                  type: bool
+                packet_length:
+                  type: dict
+                  mutually_exclusive: [[eq, lt, neq, range], [eq, gt, neq, range]]
+                  suboptions:
+                    eq:
+                      description: Match only packets on a given packet length
+                      type: int
+                    gt:
+                      description: Match only packets with a greater packet length
+                      type: int
+                    lt:
+                      description: Match only packets with a lower packet length
+                      type: int
+                    neq:
+                      description: Match only packets not on a given packet length
+                      type: int
+                    range:
+                      description: Match only packets in the range of packet lengths
+                      type: dict
+                      suboptions:
+                        start:
+                          description: Start of the packet length range
+                          type: int
+                        end:
+                          description: End of the packet length range
+                          type: int
+                precedence:
+                  description: Match packets with given precedence value
+                  type: str
+                ttl:
```

Comment:
dscp, fragments, ttl and tos should be the list of diff param or should be separate param

Reply:
You mean should these be top level keys or under another key called `match`?